cblox/include/cblox/core/submap_collection.h

```diff
@@ -0,0 +1,138 @@
+#ifndef CBLOX_CORE_SUBMAP_COLLECTION_H_
+#define CBLOX_CORE_SUBMAP_COLLECTION_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "./TsdfSubmapCollection.pb.h"
+#include "cblox/core/common.h"
+#include "cblox/core/tsdf_esdf_submap.h"
+
+namespace cblox {
+
+template <typename SubmapType>
+class SubmapCollection {
+ public:
+  typedef std::shared_ptr<SubmapCollection> Ptr;
+  typedef std::shared_ptr<const SubmapCollection> ConstPtr;
+
+  // Constructor. Constructs an empty submap collection map
+  explicit SubmapCollection(const typename SubmapType::Config &submap_config)
+      : submap_config_(submap_config) {}
+
+  // Constructor. Constructs a submap collection from a list of submaps
+  SubmapCollection(const typename SubmapType::Config &submap_config,
+                   const std::vector<typename SubmapType::Ptr> &tsdf_sub_maps);
+
+  // Gets a vector of the linked IDs
+  std::vector<SubmapID> getIDs() const;
+  bool exists(const SubmapID submap_id) const;
+
+  // Creates a new submap on the top of the collection
+  void createNewSubMap(const Transformation &T_M_S, const SubmapID submap_id);
```

Comment:
It's probably better design to return a bool indicating the submap already exists and the function has not succeeded. In our use case, this would be a bug, though so let's leave the hard fail in for now.